circus-train-avro/src/main/java/com/hotels/bdp/circustrain/avro/conf/AvroSerDeConfig.java

```diff
@@ -33,4 +36,12 @@ public String getBaseUrl() {
   public void setBaseUrl(String avroSchemaBaseUrl) {
     this.avroSchemaBaseUrl = avroSchemaBaseUrl;
   }
+
+  public String getSourceAvroUrlScheme() {
```

Comment:
Can you not get the scheme from the configured hiveConf? path.getFileSystem(sourceHiveConf).getScheme()?

Reply:
I'm not sure if this is a good idea with how we transform the FileSystem in Circus Train, getScheme is implementation specific and change the FileSystem depending on where the source and destination of the replication resides. Here is the getScheme methods for DistributedFileSystem, S3FileSystem and GoogleHadoopFileSystem respectively:

``` java
public String getScheme() {
    return "hdfs";
}

public String getScheme() {
    return "s3";
}

public String getScheme() {
    return "gs";
}
```

Reply:
That is my point depending on the config we get the correct FileSystem and thus the correct scheme. Why is the example you give wrong?

Reply:
I'm not sure that's the case with how we're binding the filesystems, for example say your source is in Chandler and destination is in S3 we would bind the S3FileSystem. In Chandler an avro.schema.url with no scheme will default to HDFS, but if we bind the S3FileSystem in an initialization step which I believe we do then the default scheme becomes S3 and the file won't be found. I will test it I'm not 100% sure on the ordering of things and when certain things are triggered

Reply:
Please check. My understanding is that you still get a DistributedFileSystemt back. 
The binding does: fs.s3.impl=foo.bar.S3FileSystem.class. Which means if you need to load a S3 FileSystem use this particular implementation. It doesn't say if you create a fileSystem always create an S3FileSystem.
Do you really get an S3FileSystem back?

Reply:
Hmm it might fall back to "file" scheme (unless this is set: 'fs.defaultFS'), would be interesting to see what you find out.